crates/wasmtime/src/store.rs

```diff
@@ -1251,11 +1310,15 @@ impl StoreOpaque {
 
     #[cfg(feature = "async")]
     #[inline]
-    pub fn async_cx(&self) -> AsyncCx {
+    pub fn async_cx(&self) -> Option<AsyncCx> {
```

Comment:
Could you add some documentation to this function as to why `Option` is the return value? (notably pointing out as well that this panics if `async_support` is not enabled, whereas the signature might naively appear to return `None` if async support is not enabled)

Reply:
Is there any reason not to just make the check be for both conditions (i.e., we have `async_support` and we're not being torn down), rather than panicking if `async_support` is not enabled?

Reply:
(I've added the basic docs, with the panic sill allowed, in 09d62a4)